source/PyMaterialX/CMakeLists.txt

```diff
@@ -47,16 +47,22 @@ endif()
 
 add_subdirectory(PyMaterialXCore)
 add_subdirectory(PyMaterialXFormat)
-add_subdirectory(PyMaterialXGenShader)
-add_subdirectory(PyMaterialXGenGlsl)
-add_subdirectory(PyMaterialXGenOsl)
-if (MATERIALX_BUILD_GEN_OGSFX)
-    add_subdirectory(PyMaterialXGenOgsFx)
-endif()
-if (MATERIALX_BUILD_GEN_ARNOLD)
-    add_subdirectory(PyMaterialXGenArnold)
+if (MATERIALX_BUILD_GEN)
```

Comment:
We'll need to check whether ILM want's this back as originally we had all GLSL and OSL modules as required so would never do this check. I think it should be okay or we we split along "sets" of requirements. e.g. desktop always has GLSL + OSL + MDL. We can figure this out.

Reply:
This is central to our use case -- we need to be able to build without gen at all, and later, to build just with GLSL gen.